### PR TITLE
Disc update

### PR DESCRIPTION
**Pull request by taherbert**
Updating a few pieces of logic and cleaning up some of the spell data. Also adds a few missing spells.

### CODE REVIEW

File: Dragonflight/PriestDiscipline.lua [hunk 1]
```diff
@@ -145,7 +144,7 @@ spec:RegisterPvpTalents( {
 
 -- Auras
 spec:RegisterAuras( {
-    apathy = {
+        apathy = {
```
Comment (Hekili) [inline Dragonflight/PriestDiscipline.lua]: Fix indentation.

File: Dragonflight/PriestDiscipline.lua [hunk 2]
```diff
@@ -160,7 +159,7 @@ spec:RegisterAuras( {
         duration = 15,
         max_stack = 1
     },
-    body_and_soul = {
+        body_and_soul = {
```
Comment (Hekili) [inline Dragonflight/PriestDiscipline.lua]: Fix indentation.

File: Dragonflight/PriestDiscipline.lua [hunk 3]
```diff
@@ -176,6 +175,11 @@ spec:RegisterAuras( {
         max_stack = 1
     },
     death_and_madness = {
+        id = 321973,
```
Comment (Hekili) [inline Dragonflight/PriestDiscipline.lua]: 321973 is not a visible/interactive aura from an API standpoint.  Why is this needed vs. what is included as death_and_madness_debuff already?

File: Dragonflight/PriestDiscipline.lua [hunk 4]
```diff
@@ -209,7 +213,7 @@ spec:RegisterAuras( {
     harsh_discipline = {
         id = 373181,
         duration = 3600,
-        max_stack = 10
+        max_stack = function() return 8/talent.harsh_discipline.rank end,
```
Comment (Hekili) [inline Dragonflight/PriestDiscipline.lua]: Formatting:
function() return 8 / talent.harsh_discipline.rank end,

File: Dragonflight/PriestDiscipline.lua [hunk 5]
```diff
@@ -309,12 +314,12 @@ spec:RegisterAuras( {
         tick_time = 3,
         max_stack = 1
     },
-    schism = {
+        schism = {
```
Comment (Hekili) [inline Dragonflight/PriestDiscipline.lua]: Fix indent.

File: Dragonflight/PriestDiscipline.lua [hunk 6]
```diff
@@ -309,12 +314,12 @@ spec:RegisterAuras( {
...
         id = 214621,
-        duration = 9,
+        duration = function() return talent.malicious_intent.enabled and 15 or 9 end,
         max_stack = 1
     },
-    shackle_undead = {
+        shackle_undead = {
```
Comment (Hekili) [inline Dragonflight/PriestDiscipline.lua]: Fix indent.

File: Dragonflight/PriestDiscipline.lua [hunk 7]
```diff
@@ -454,19 +483,20 @@ spec:RegisterAbilities( {
...
+        school = function() return buff.shadow_covenant.up and "shadow" or "holy" end,
         damage = 1,
 
         spend = 0.02,
         spendType = "mana",
 
         talent = "divine_star",
         startsCombat = true,
-        texture = 537026,
+        texture = function() return buff.shadow_covenant.up and 122121 or 110744 end,
```
Comment (Hekili) [inline Dragonflight/PriestDiscipline.lua]: These aren't texture IDs.

File: Dragonflight/PriestDiscipline.lua [hunk 8]
```diff
@@ -555,40 +606,96 @@ spec:RegisterAbilities( {
...
         id = 8092,
-        cast = 1.5,
+        cast = function() return 1.5 * haste end,
         charges = function()
             if talent.dark_indulgence.enabled then return 2 end
         end,
         cooldown = 9,
         recharge = function()
-            if talent.dark_indulgence.enabled then return 9 end
+            if talent.dark_indulgence.enabled then return 9*haste end
```
Comment (Hekili) [inline Dragonflight/PriestDiscipline.lua]: Formatting:
`if talent.dark_indulgence.enabled then return 9 * haste end`

File: Dragonflight/PriestDiscipline.lua [hunk 9]
```diff
@@ -555,40 +606,96 @@ spec:RegisterAbilities( {
...
         spendType = "mana",
 
         startsCombat = true,
         texture = 136224,
 
         handler = function ()
             if talent.harsh_discipline.enabled then
-                if buff.harsh_discipline.up and buff.harsh_discipline.stack == 9 then
+                addStack( "harsh_discipline" )
+                if buff.harsh_discipline.up and buff.harsh_discipline.stack == 8/talent.harsh_discipline.rank then
```
Comment (Hekili) [inline Dragonflight/PriestDiscipline.lua]: `if buff.harsh_discipline.up and buff.harsh_discipline.stack == buff.harsh_discipline.max_stack - 1 then`

File: Dragonflight/PriestDiscipline.lua [hunk 10]
```diff
@@ -555,40 +606,96 @@ spec:RegisterAbilities( {
...
+                    else
+                        debuff.shadow_word_pain.expires = debuff.shadow_word_pain.expires - 6
+                    end
                 end
             end
+        end,
+    },
 
-            if talent.inescapable_torment.enabled and pet.fiend.active then pet.fiend.expires = pet.fiend.expires + 1 end
+    mindbender = {
```
Comment (Hekili) [inline Dragonflight/PriestDiscipline.lua]: Do not add a Discipline version of Mindbender.  You can add Discipline specific considerations to mindbender in PriestShadow.lua if needed.

File: Dragonflight/PriestDiscipline.lua [hunk 11]
```diff
@@ -555,40 +606,96 @@ spec:RegisterAbilities( {
...
-            if talent.void_summoner.enabled then reduceCooldown( "mindbender", 2 ) end
+        toggle = "cooldowns",
+
+        handler = function ()
+            summonPet( "mindbender", 15 )
+            applyBuff( "mindbender" )
         end,
     },
 
+    mindgames = {
```
Comment (Hekili) [inline Dragonflight/PriestDiscipline.lua]: Does not need to be duplicated if provided in another Priest module.

File: Dragonflight/PriestDiscipline.lua [hunk 12]
```diff
@@ -631,15 +739,56 @@ spec:RegisterAbilities( {
...
             if debuff.purge_the_wicked.up then active_dot.purge_the_wicked = min( active_dot.purge_the_wicked + 1, true_active_enemies ) end
-
-            if talent.void_summoner.enabled then reduceCooldown( "mindbender", 2 ) end
         end,
 
         copy = { 186720, 400169, "dark_reprimand" }
     },
 
 
+    power_infusion = {
```
Comment (Hekili) [inline Dragonflight/PriestDiscipline.lua:691]: Does not need to be added if provided by another Priest specialization module.

File: Dragonflight/PriestDiscipline.lua [hunk 13]
```diff
@@ -686,8 +835,41 @@ spec:RegisterAbilities( {
             end
         end,
     },
+    -- Shields an ally for 15 sec, absorbing 25,497 damage.
```
Comment (Hekili) [inline Dragonflight/PriestDiscipline.lua]: Does not need to be added if provided by another Priest specialization module.

File: Dragonflight/PriestDiscipline.lua [hunk 14]
```diff
@@ -697,22 +879,30 @@ spec:RegisterAbilities( {
...
+                reduceCooldown( "mindgames", 0.5 * talent.manipulation.rank )
+            end
+            if talent.train_of_thought.enabled then
+                reduceCooldown( "penance", 0.5 )
+            end
             if talent.harsh_discipline.enabled then
-                if buff.harsh_discipline.up and buff.harsh_discipline.stack == 9 then
-                    removeBuff( "harsh_discipline" )
+                addStack( "harsh_discipline" )
+                if buff.harsh_discipline.stack == 8/talent.harsh_discipline.rank then
```
Comment (Hekili) [inline Dragonflight/PriestDiscipline.lua]: Like the other case, don't want to addStack to decide to removeBuff.

`if buff.harsh_discipline.stack == buff.harsh_discipline.max_stack - 1 then`

File: Dragonflight/PriestDiscipline.lua [hunk 15]
```diff
@@ -845,32 +1036,109 @@ spec:RegisterAbilities( {
         end,
     },
 
-    -- Smites an enemy for 1,315 Holy damage.
+    shadowfiend = {
```
Comment (Hekili) [inline Dragonflight/PriestDiscipline.lua]: Do not add to Discipline; it exists elsewhere.

File: Dragonflight/PriestDiscipline.lua [hunk 16]
```diff
@@ -845,32 +1036,109 @@ spec:RegisterAbilities( {
...
+                end
+            end
+        end,
+    },
+
+
     smite = {
         id = 585,
-        cast = 1.5,
+        cast = function() return 1.5*haste end,
```
Comment (Hekili) [inline Dragonflight/PriestDiscipline.lua]: This should already be hasted automatically.

Try `/dump Hekili.Class.abilities.smite.cast` in-game to verify before making this change.

File: Dragonflight/PriestDiscipline.lua [hunk 17]
```diff
@@ -845,32 +1036,109 @@ spec:RegisterAbilities( {
...
         texture = 135924,
 
         handler = function ()
+            if talent.train_of_thought.enabled then
+                reduceCooldown( "penance", 0.5 )
+            end
             if talent.harsh_discipline.enabled then
-                if buff.harsh_discipline.up and buff.harsh_discipline.stack == 9 then
+                addStack( "harsh_discipline" )
+                if buff.harsh_discipline.stack == 8/talent.harsh_discipline.rank then
```
Comment (Hekili) [inline Dragonflight/PriestDiscipline.lua]: Same as previous comments re: Harsh Discipline.

**Review by Hekili — COMMENTED** [review]
Overall, some minor formatting changes.

If a spell exists in another module and isn't significantly different, just add Discipline's mechanics to the other spell.  It helps avoid issues where spells get out of sync.  Shadowfiend and Mindbender involve some keybind swapping mechanics but otherwise share the same entry.

File: Dragonflight/PriestDiscipline.lua (hunk 3 above)
Comment (taherbert) [reply Dragonflight/PriestDiscipline.lua]: I think you're right -- removed.

File: Dragonflight/PriestDiscipline.lua [hunk 18]
```diff
@@ -533,10 +566,28 @@ spec:RegisterAbilities( {
         copy = 120644
     },
 
+    holy_nova = {
```
Comment (taherbert) [inline Dragonflight/PriestDiscipline.lua]: @Hekili removing holy nova as well, duplicative with shadow implementation

File: Dragonflight/PriestDiscipline.lua (hunk 16 above)
Comment (taherbert) [reply Dragonflight/PriestDiscipline.lua]: You're correct!

File: Dragonflight/PriestDiscipline.lua (hunk 13 above)
Comment (taherbert) [reply Dragonflight/PriestDiscipline.lua]: Removed in favor of Shadow's version, but updated to reflect that weakened soul is no longer applied.

File: Dragonflight/PriestDiscipline.lua [hunk 19]
```diff
@@ -845,32 +1036,109 @@ spec:RegisterAbilities( {
...
+        toggle = "cooldowns",
+
+        handler = function ()
+            summonPet( "shadowfiend", 15 )
+            applyBuff( "shadowfiend" )
+        end,
+    },
+
+
+    shadow_word_death = {
```
Comment (taherbert) [inline Dragonflight/PriestDiscipline.lua]: Updating Shadow's implementation instead

File: Dragonflight/PriestDiscipline.lua [hunk 20]
```diff
@@ -899,8 +1167,8 @@ spec:RegisterAbilities( {
     vampiric_embrace = {
```
Comment (taherbert) [inline Dragonflight/PriestDiscipline.lua]: Removed in favor of Shadow's implementation

**Comment by taherbert** [comment]
@Hekili thanks for the feedback, addressed your comments! Moved some of the logic to Shadow's implementation instead and removed some abilities that are unlikely to ever be used in any kind of rotation (e.g. Pain Suppression).

**Comment by Hekili** [comment]
> @Hekili thanks for the feedback, addressed your comments! Moved some of the logic to Shadow's implementation instead and removed some abilities that are unlikely to ever be used in any kind of rotation (e.g. Pain Suppression).

PI can stay; it's conceivable somebody might put it as a self-defensive or something in a custom priority.

**Comment by taherbert** [comment]
> PI can stay

Sure, re-added @Hekili

**Review by Hekili — COMMENTED** [review]
I made a few changes to compatibility across specs.
